lib/requester/util.js

```diff
@@ -116,29 +118,37 @@ module.exports = {
      *
      * @todo: Move this to the SDK.
      * @param request
+     * @param options
      */
-    getRequestBody: function (request) {
-        var options,
+    getRequestBody: function (request, options) {
+        var reqBody,
             mode = _.get(request, 'body.mode'),
             body = _.get(request, 'body'),
             empty = request.body && request.body.isEmpty && request.body.isEmpty(),
+            method,
             content;
 
         if (empty || !mode || !body[mode]) {
             return;
         }
 
+        // Do not send a body if the request is GET and sendBodyWithGetRequests is falsey.
+        method = _.isFunction(_.get(request, 'method.toLowerCase')) ? request.method.toLowerCase() : undefined;
```

Comment:
can we pre-check `request` on top and do early bail out? When done, then this code will be

```
method = _.isString(request.method) ? request.method.toLowerCase() : undefined;
```